File: tools/cli/cmd/launch.cpp
#include "launch.hpp"

namespace bad::cli::cmd::launch {

void run(const Config &config) {
  std::cout << "LAUNCH" << std::endl;
  std::cout << "PROGRAM: " << config.program_path << std::endl;
}

void setup(CLI::App &app) {
  auto config = std::make_shared<Config>();
  auto cmd = app.add_subcommand("launch", "Launch a program.");
  cmd->add_option("program_path", config->program_path,
                  "Path of program to launch.")
      ->required();
  cmd->callback([config]() { run(*config); });
}

} // namespace bad::cli::cmd::launch

File: bad.rs/bad-cli/src/cmd/launch.rs
use std::path::PathBuf;

pub struct Config {
    pub program_path: PathBuf,
}

pub fn run(config: &Config) -> anyhow::Result<()> {
    println!("LAUNCH");
    println!("PROGRAM: {}", config.program_path.display());
    Ok(())
}
